druid-shell/src/platform/gtk/window.rs

```diff
@@ -361,6 +366,33 @@ impl WindowBuilder {
             );
         }
 
+        win_state
+            .drawing_area
+            .connect_realize(clone!(handle => move |drawing_area| {
+                if let Some(clock) = drawing_area.get_frame_clock() {
+                    clock.connect_before_paint(clone!(handle => move |_clock|{
+                        if let Some(state) = handle.state.upgrade() {
+                            state.in_draw.set(true);
+                        }
+                    }));
+                    clock.connect_after_paint(clone!(handle => move |_clock|{
+                        if let Some(state) = handle.state.upgrade() {
+                            state.in_draw.set(false);
+                            if state.request_animation.get() {
+                                state.request_animation.set(false);
+                                let vbox = state.window.get_children().first().unwrap().clone()
+                                    .downcast::<gtk::Box>()
+                                    .unwrap();
+                                let first_child = vbox.get_children().last().unwrap().clone();
+                                if first_child.is::<gtk::DrawingArea>() {
+                                    first_child.queue_draw();
```

Comment:
I think we can also use state.drawing_area here? Sorry for missing this earlier

Reply:
I also looked over everything to see if it could be simplified more, and also missed it. NP just pushed this too.